magento/magento2
# Fix ui switcher element not cleaned up on input destroy

### Description (*)

The ui switcher element is created by the ui form field element if it is enabled, and is added to the uiRegistry when it is initialized.
If the form field is then destroyed (either manually or from a `destroyInserted` of an `insert-form`) the switcher is not removed from the uiRegistry.

This causes an issue if that form field is created again, as the previous switcher still exists and a new instance is not created, causing it to not be linked to the new form field.

### Related Pull Requests


### Fixed Issues (if relevant)

fixes magento/magento2#39553

### Manual testing scenarios (*)

1. Create two ui component forms.
2. In the first form, create a `button` component, and a `modal` component with an `insert-form` component.
3. Set the button's actions to `destroyInserted` on the insert-form, `openModal` on the modal, and `render` on the insert-form.
4. Set the insert-form to render second form, and set `autoRender` to false.
5. Add a checkbox field (default unchecked) and an input field to the second form.
6. Set up `switcherConfig` for the checkbox field so that it enables/disables the input field when the checkbox is toggled.
7. Open the first form and click the button to open the modal with the second form.
8. Toggle the checkbox to enable the input field and enter a value, then close the modal.
9. Open the modal again and toggle the checkbox. The input field should be enabled/disabled as expected.

### Questions or comments


### Contribution checklist (*)
